dataflux_pytorch/integration_tests/integration_test.py

```diff
@@ -0,0 +1,193 @@
+            if config["max_compose_bytes"]
+            else 100000000
+        )
+        if config["list_timeout"]:
+            config["list_timeout"] = float(config["list_timeout"])
+        if config["download_timeout"]:
+            config["download_timeout"] = float(config["download_timeout"])
+        config["parallelization"] = (
+            int(config["parallelization"]) if config["parallelization"] else 1
+        )
+
+        return config
+
+    def test_list_and_load_iter(self):
+        list_start_time = time.time()
+        test_config = self.get_config()
+        config = dataflux_iterable_dataset.Config()
+        if test_config["prefix"]:
+            config.prefix = test_config["prefix"]
+
+        # Define the data_format_fn to transform the data samples.
+        # NOTE: Make sure to modify this to fit your data format.
+        def read_image_modified(content_in_bytes):
+            return numpy.load(io.BytesIO(content_in_bytes), allow_pickle=True)["x"]
+
+        dataset = dataflux_iterable_dataset.DataFluxIterableDataset(
+            project_name=test_config["project"],
+            bucket_name=test_config["bucket"],
+            config=config,
+            data_format_fn=read_image_modified,
+        )
+        list_end_time = time.time()
+        listing_time = list_end_time - list_start_time
+        if (
+            test_config["expected_file_count"]
+            and len(dataset.objects) != test_config["expected_file_count"]
+        ):
+            raise AssertionError(
+                f"Expected {test_config['expected_file_count']} files, but got {len(dataset.objects)}"
+            )
+        if test_config["list_timeout"] and listing_time > test_config["list_timeout"]:
+            raise AssertionError(
+                f"Expected list operation to complete in under {test_config['list_timeout']} seconds, but took {listing_time} seconds."
+            )
+        data_loader = data.DataLoader(
+            dataset=dataset,
+            batch_size=100,
+            num_workers=test_config["parallelization"],
+            prefetch_factor=2,
+            persistent_workers=True,
+            pin_memory=True,
+        )
+        training_start_time = time.time()
+        for i in range(1):
+            total_objects = 0
+            total_bytes = 0
+            epoch_start = time.time()
+            last_update = time.time()
+            for batch in data_loader:
+                # A simple sleep function to simulate the GPU training time.
+                time.sleep(0.1)
+
+                total_objects += len(batch)
+                for object_bytes in batch:
+                    total_bytes += len(object_bytes)
+                if time.time() - last_update > 5:
+                    last_update = time.time()
+            epoch_end = time.time()
+        training_end_time = time.time()
+        if (
+            test_config["download_timeout"]
+            and training_time > test_config["download_timeout"]
```

Comment:
I don't think I see the `training_time` calculation in this function? I see it in the one below, though